openedx/frontend-app-authoring
# feat: collections page [FC-0062]

## Description

Adds collections page as described in https://github.com/openedx/frontend-app-course-authoring/issues/1101

Also addresses following issues.

* https://github.com/openedx/frontend-app-authoring/issues/1103
* https://github.com/openedx/frontend-app-authoring/issues/1169
* https://github.com/openedx/frontend-app-authoring/issues/1168
* https://github.com/openedx/frontend-app-authoring/issues/1167

Depends on https://github.com/openedx/edx-platform/pull/35493

## Supporting information

* Issue: https://github.com/openedx/frontend-app-course-authoring/issues/1101
* `Private-ref`: [FAL-3790](https://tasks.opencraft.com/browse/FAL-3790)
* https://github.com/openedx/edx-platform/pull/35493

## Testing instructions


* Checkout https://github.com/openedx/edx-platform/pull/35493
* Setup tutor with this PR and https://github.com/open-craft/tutor-contrib-meilisearch plugin
* Make sure to create few libraries
* Open collections tab in a library, add new collection using new button.
* On save, collections page should open up with no components.
* Add some components and verify that they are associated with the current collection and visible.
* Test features from issues listed above.

## Concerns

* The collections page is only accessible from UI when user creates a new one, i.e. users cannot click on a collection and see collections page as it seems like the collections sidebar should open on click and not the page.
